src/transformStateWithClones.js

```diff
@@ -1,13 +1,31 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const current = { ...state };
+  const resultingArray = [];
+  let newr;
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const prop in actions[i].extraData) {
+        current[prop] = actions[i].extraData[prop];
+      }
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const word in actions[i].keysToRemove) {
```

Comment:
Why do you use 3 different words for the same thing `prop`, `word` and `key`?

Reply:
I don't know. I didn't think it was important at this point, since those variables were used only in those Individual loops...
I`ll change it